### DISCUSSION TITLE
I need an idea on how to automatically create a branch from an existing solution in Visual Studio.

### DISCUSSION THREAD

**Discussion by Aditya-Agarwal01** [community/community#137068] [Programming Help]
### Body

We have several .NET Core solution files, each containing multiple projects. Our task is to create a new branch for each solution, using an already existing solution as a parent in Visual Studio. These projects are connected to Team Explorer through TFS.

Can this process be automated using a batch or PowerShell script? If so, how can it be done?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Yackens** [comment]
Yes, you can automate the process of creating a new branch for each solution using a PowerShell script. The script will interact with Git via the command line to create branches and switch to them. If your solutions are connected to TFS, the script can also handle those connections.

```
# Define the parent branch
$parentBranch = "parent-branch-name"

# Define the new branch name
$newBranchName = "new-branch-name"

# Define the path to your solutions
$solutionPaths = @(
    "C:\Path\To\Solution1",
    "C:\Path\To\Solution2",
    "C:\Path\To\Solution3"
)

foreach ($path in $solutionPaths) {
    Write-Host "Processing solution at $path"

    # Navigate to the solution directory
    Set-Location $path

    # Fetch the latest changes from the parent branch
    git fetch origin $parentBranch

    # Create and check out a new branch from the parent branch
    git checkout -b $newBranchName origin/$parentBranch

    Write-Host "Created and checked out branch $newBranchName for solution at $path"
}

# Navigate back to the original location
Set-Location $PSScriptRoot

Write-Host "Branch creation process completed."
```

Replace "parent-branch-name" with the name of the existing parent branch.
Replace "new-branch-name" with the name you want for the new branch.
Replace the paths in $solutionPaths with the actual paths to your solution directories.


If it is useful, don't forget to mark as the accepted answer so more people can see it 😄

**Reply by Aditya-Agarwal01** [reply]
Thank you for the suggestion, but it won't work for .NET Core code connected to TFS in Visual Studio. The execution completes successfully, but no actions are performed.😊
